youtube_dl/extractor/nieuwsblad.py

```diff
@@ -0,0 +1,93 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+
+from .common import InfoExtractor
+from ..utils import (
+    smuggle_url
+)
+
+
+class NieuwsbladIE(InfoExtractor):
+    """ Extractor for www.nieuwsblad.be """
+    _VALID_URL = r'https?://(?:www\.)?nieuwsblad\.be/.+?/dmf([0-9]+?)_(?P<id>[0-9]+)'
+    _TESTS = [
+        # Source: VMMA
+        {
+            'url': 'http://www.nieuwsblad.be/cnt/dmf20151224_02036890',
+            'md5': '3dcf2c3a140d8e54dd8376d4c4a609f4',
+            'info_dict': {
+                'id': '02036890',
+                'ext': 'mp4',
+                'title': 'Krijgt zieke Pauline (3) het mooiste kerstcadeau?',
+                'thumbnail': 're:http.*jpg$',
+            }
+        },
+        # Source: VRT
+        {
+            'url': 'http://www.nieuwsblad.be/cnt/dmf20151124_01986463',
+            'md5': '8e46cb7ddfddeb64735fa39f105002c2',
+            'info_dict': {
+                'id': '01986463',
+                'ext': 'mp4',
+                'title': 'Angst voor terreur: fotograaf toont hoe hij de werkelijkheid kan manipuleren',
+                'thumbnail': 're:http.*jpg$',
+            }
+        },
+        # Source: Mediahuis (using kaltura)
+        {
+            'url': 'http://www.nieuwsblad.be/cnt/dmf20151225_02037264',
+            'md5': 'd4decdc7f105c26767b928c54c7d5184',
+            'info_dict': {
+                'id': '1_z4jndqki',
+                'ext': 'mov',
+                'title': 'autobrand Peer',
+                'thumbnail': 're:^https?://.*/thumbnail/.*',
+                'timestamp': int,
+                'upload_date': '20151225',
+                'uploader_id': 'dcc-video-manager-hbvl@mediahuis.be'
+            }
+        },
+    ]
+
+    def _real_extract(self, url):
+        """ Extract the video info from the given 'nieuwsblad' URL """
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        iframe_m = re.search(r'<script[^>]+src="(.+?kaltura.com.*?)"', webpage)
+        if iframe_m:
+            return self._extract_kaltura(url, webpage)
+
+        thumbnail = self._og_search_thumbnail(webpage)
+        title = self._og_search_title(webpage)
+
+        iframe_m = re.search(r'<iframe[^>]+src="(.+?vrt.be.*?)"', webpage)
+        if iframe_m:
+            webpage = self._download_webpage(iframe_m.group(1), "vrt-iframe")
+            video_url = self._search_regex(r'sources.pdl = "(.*?)";', webpage, 'vrt-video')
+
+        iframe_m = re.search(r'<iframe[^>]+src="(.+?vmma.be.*?)"', webpage)
+        if iframe_m:
+            webpage = self._download_webpage(iframe_m.group(1), "vmma-iframe")
+            video_url = self._search_regex(r'<source src="(.*?)"', webpage, 'vmma-video')
```

Comment:
Video IDs are used in --download-archive. As long it's unique across videos belonging to the same extractor (i.e., different videos always map to different IDs), it's a good choice. If there are multiple satisfying IDs, it's up to you.